pkg/util/k8s/k8s.go

```diff
@@ -1836,3 +1836,173 @@ func GetV1beta1CRDFromFile(
 	}
 	return crd, nil
 }
+
+// GetAllObjects gets all objects from k8s
+func GetAllObjects(k8sClient client.Client, namespace string) ([]client.Object, error) {
+	var objs []client.Object
+
+	if err := AppendObjectList(k8sClient, namespace, &v1.ServiceList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &v1.ServiceAccountList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &v1.SecretList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &v1.ConfigMapList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &v1.PodList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &v1.PersistentVolumeClaimList{}, &objs); err != nil {
+		return objs, err
+	}
+
+	if err := AppendObjectList(k8sClient, namespace, &appsv1.DaemonSetList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &appsv1.DeploymentList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &appsv1.StatefulSetList{}, &objs); err != nil {
+		return objs, err
+	}
+
+	if err := AppendObjectList(k8sClient, "", &rbacv1.ClusterRoleList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, "", &rbacv1.ClusterRoleBindingList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &rbacv1.RoleList{}, &objs); err != nil {
+		return objs, err
+	}
+	if err := AppendObjectList(k8sClient, namespace, &rbacv1.RoleBindingList{}, &objs); err != nil {
+		return objs, err
+	}
+
+	if err := AppendObjectList(k8sClient, "", &storagev1.StorageClassList{}, &objs); err != nil {
+		return objs, err
+	}
+
+	return objs, nil
+}
+
+// AppendObjectList gets objects from k8s and adds to the list
+func AppendObjectList(k8sClient client.Client, namespace string, list client.ObjectList, objs *[]client.Object) error {
+	err := k8sClient.List(
+		context.TODO(),
+		list,
+		&client.ListOptions{
+			Namespace: namespace,
+		})
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to list objects %v", list.GetObjectKind())
+		return err
+	}
+
+	if l, ok := list.(*v1.ServiceList); ok {
```

Comment:
Can we change this to be a loop?

Reply:
looks like this type conversion is hard to do in list, but i changed the other one into loop